starcounter-include.html

```diff
@@ -32,13 +32,74 @@
             static get observedAttributes() {
                 return ['partial', 'view-model', 'partialId'];
             }
-            // the self argument might be provided or not
-            // in both cases, the mandatory `super()` call
-            // will return the right context/instance to use
-            // and eventually return
-            constructor(self) {
-                self = super(self);
-                this.createComponent();
+            /**
+            * Create shadowRoot, define property setters, set initial partial.
+            */
+            constructor() {
+                super();
+                this.defaultComposition = null;
+                var partialId = this.partialId;
+                var partial = this.viewModel || this.partial || undefined;
+                checkForNonNamespaced(partial, this);
+
+                // build shadow DOM
+                if(useShadowDOMV1){
+                    this.attachShadow({mode: "open"});
+                } else {
+                    this.createShadowRoot();
+                }
+                // stamp composition
+                // like this.stampComposition() without an event
+                this.shadowRoot.innerHTML = defaultShadowCSS + defaultShadowDOM;
```

Comment:
Do you think we could live without this line? It is a very slow operation that will most probably be replaced anyway.

Reply:
How could you be sure that it will be replaced and will not be needed?

Reply:
It seems to me that this line is only useful if a `partial` is not provided to `starcounter-include`, which should never happen.

If `partial` is provided, Shadow DOM will be replaced by `updateComposition`. If partial does not have a composition, it will render the fallback composition at the line: https://github.com/Starcounter/starcounter-include/blob/020bea936c4880bd9b4c761fbbff795793c66d03/starcounter-include.html#L295

Reply:
> It seems to me that this line is only useful if a partial is not provided to starcounter-include, which should never happen.

Reply:
OK, let's get back to this in future when doing performance improvements.